internal/persistence/persistors/mysqlstore/store_organization.go

```diff
@@ -0,0 +1,290 @@
+			fmt.Sprintf("%s.%s AS %s",
+				mysqlmodel.TableNames.Organization,
+				mysqlmodel.OrganizationColumns.Name,
+				mysqlmodel.OrganizationColumns.Name,
+			),
+			fmt.Sprintf("%s.%s AS %s",
+				mysqlmodel.TableNames.Organization,
+				mysqlmodel.OrganizationColumns.CreatedBy,
+				mysqlmodel.OrganizationColumns.CreatedBy,
+			),
+			fmt.Sprintf("%s.%s AS %s",
+				mysqlmodel.TableNames.Organization,
+				mysqlmodel.OrganizationColumns.LastUpdatedBy,
+				mysqlmodel.OrganizationColumns.LastUpdatedBy,
+			),
+			fmt.Sprintf("%s.%s AS %s",
+				mysqlmodel.TableNames.Organization,
+				mysqlmodel.OrganizationColumns.CreatedAt,
+				mysqlmodel.OrganizationColumns.CreatedAt,
+			),
+			fmt.Sprintf("%s.%s AS %s",
+				mysqlmodel.TableNames.Organization,
+				mysqlmodel.OrganizationColumns.LastUpdatedAt,
+				mysqlmodel.OrganizationColumns.LastUpdatedAt,
+			),
+			fmt.Sprintf("%s.%s AS %s",
+				mysqlmodel.TableNames.Organization,
+				mysqlmodel.OrganizationColumns.IsActive,
+				mysqlmodel.OrganizationColumns.IsActive,
+			),
+		),
+	}
+
+	if filters != nil {
+		if len(filters.IdsIn) > 0 {
+			queryMods = append(queryMods, mysqlmodel.OrganizationWhere.ID.IN(filters.IdsIn))
+		}
+
+		if filters.OrganizationIsActive.Valid {
+			queryMods = append(queryMods, mysqlmodel.OrganizationWhere.IsActive.EQ(filters.OrganizationIsActive.Bool))
+		}
+
+		if len(filters.OrganizationNameIn) > 0 {
+			queryMods = append(queryMods, mysqlmodel.OrganizationWhere.Name.IN(filters.OrganizationNameIn))
+		}
+
+		if filters.CreatedBy.Valid {
+			queryMods = append(queryMods, mysqlmodel.OrganizationWhere.CreatedBy.EQ(filters.CreatedBy))
+		}
+
+		if filters.LastUpdatedBy.Valid {
+			queryMods = append(queryMods, mysqlmodel.OrganizationWhere.LastUpdatedBy.EQ(filters.LastUpdatedBy))
+		}
+	}
+
+	q := mysqlmodel.Organizations(queryMods...)
+	totalCount, err := q.Count(ctx, ctxExec)
+	if err != nil {
+		return nil, fmt.Errorf("get organizations count: %w", err)
+	}
+
+	page := pagination.Page
+	maxRows := pagination.MaxRows
+	if filters != nil {
+		if filters.Page.Valid {
+			page = filters.Page.Int
+		}
+		if filters.MaxRows.Valid {
+			maxRows = filters.MaxRows.Int
+		}
+	}
+
+	pagination.SetQueryBoundaries(page, maxRows, int(totalCount))
+
+	queryMods = append(queryMods, qm.Limit(pagination.MaxRows), qm.Offset(pagination.Offset))
+	q = mysqlmodel.Organizations(queryMods...)
+
+	if err = q.Bind(ctx, ctxExec, &res); err != nil {
+		return nil, fmt.Errorf("get organizations: %w", err)
+	}
+
+	pagination.RowCount = len(res)
+	paginated.Organizations = res
+	paginated.Pagination = pagination
+
+	return &paginated, nil
+}
+
+func (m *Repository) CreateOrganization(ctx context.Context, tx persistence.TransactionHandler, organization *model.Organization) (*model.Organization, error) {
+	ctxExec, err := mysqltx.GetCtxExecutor(tx)
+	if err != nil {
+		return nil, fmt.Errorf("extract context executor: %w", err)
+	}
+
+	entry := mysqlmodel.Organization{
+		Name: organization.Name,
+	}
```

Comment:
You need to insert the user who created the organization.
add a new parameter in the function before organization called _userId int_,
and use that!

Reply:
Ah ok ok got it.

Reply:
Oh I saw you already have it here _[func (m *Repository) AddOrganization]_, so delete this func, and modify that func.. OR, modify this and delete that.. your call.

Reply:
Do we need to create that new user id int? The createdby field is already a FK to the user id

Reply:
yeah, in your tests, whatever userID you pass should be a valid user id